manager/allocator/networkallocator/networkallocator.go

```diff
@@ -633,12 +627,31 @@ func (na *NetworkAllocator) resolveDriver(n *api.Network) (driverapi.Driver, str
 
 	d, _ := na.drvRegistry.Driver(dName)
```

Comment:
We need to check even in the case where the driver is already in driver registry because the current logic will only check the first time when the plugin is not loaded. Subsequently when the same driver name is used it will not check for the scope since it is already there in driver registry. I'd suggest moving the scope checking logic out of the `d == nil {` clause below and do it afterward so that it applies for both first time plugin init and subsequent references.